lib/database_validations/lib/validations.rb

```diff
@@ -6,6 +6,28 @@ module Validations
       alias_method :validate, :valid?
     end
 
+    class_methods do
+      def create_or_find_by!(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      def create_or_find_by(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      private
+
+      def rescue_create_or_find_by_uniqueness_exception(err, attributes)
+        raise err unless err.cause&.is_a?(ActiveRecord::RecordNotUnique)
```

Comment:
Maybe add this into `rescue_from_database_exceptions(validate, &block)` along with `ActiveRecord::InvalidForeignKey` handling?